app/code/core/Mage/Log/Model/Visitor.php

```diff
@@ -105,7 +105,7 @@ protected function _construct()
         $ignoreAgents = $this->_config->getNode('global/ignore_user_agents');
         if ($ignoreAgents) {
             $ignoreAgents = $ignoreAgents->asArray();
-            if (in_array($userAgent, $ignoreAgents)) {
+            if (in_array($userAgent, $ignoreAgents) || stristr($userAgent, $ignoreAgents)) {
```

Comment:
does this work for you? because $ignoreAgents is an array but the second parameter of `stristr` has to be a string. and shouldn't the first argument be the haystack?

Reply:
Yes, I apologize for rushing to propose a solution without testing it. I have updated the code by removing the exact match search in the array and replacing it with a string search.